### PR TITLE
feat(core): early panic if the PNG icon is not RGBA, closes #6706

### LINKED ISSUE #6706: [bug] Tauri mistakenly handles PNG24 icon
### Describe the bug

When use a 24bit PNG file as icon, the program will panic:
```
 `thread 'main' panicked at 'data.len() must fit the width, height, and row_stride', /gdk-pixbuf-0.15.11/src/pixbuf.rs:52:13`
```

TAURI treats any PNG files as RGBA, and that is incorrect.

https://github.com/tauri-apps/tauri/blob/dev/core/tauri-codegen/src/context.rs#L534

To fix this problem, we can get the color type of PNG icon by `reader.output_color_type()`, but the `png` crate does not provide an API for transforming other color types to RGBA. We should use crate `image` for the transformation.

Or just simply add a assert here, outputting some friendly information if the icon is not RGBA.

### Reproduction

_No response_

### Expected behavior

_No response_

### Platform and versions

```text
Environment
  › OS: Debian 11 X64
  › Node.js: 16.19.1
  › npm: 8.19.3
  › pnpm: 7.26.2
  › yarn: 1.22.19
  › rustup: 1.25.2
  › rustc: 1.67.1
  › cargo: 1.67.1
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
  › @tauri-apps/cli [NPM]: 1.2.3
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by meowtec**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by meowtec** [comment]
Why the issue #6706 is not linked?

**Comment by amrbashir** [comment]
PR titles don't auto-link issues, but once we merge the PR, it will create a comment with the PR title and it will auto-close it.

File: .changes/early-panic-for-png-not-rgba.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"tauri": patch
```
Comment (lucasfernog) [inline .changes/early-panic-for-png-not-rgba.md]:
```suggestion
"tauri-codegen": 'patch:enhance'
```

**Review by lucasfernog — APPROVED** [review]

**Comment by lucasfernog** [comment]
@meowtec we're now enforcing commit signing, can you rebase this PR? :heart_decoration:

**Comment by meowtec** [comment]
> @meowtec we're now enforcing commit signing, can you rebase this PR? 💟

done

**Pull request merged**